app/javascript/packs/mock-device-profiling.tsx

```diff
@@ -0,0 +1,124 @@
+import { render } from 'react-dom';
+import { useInstanceId } from '@18f/identity-react-hooks';
+import { ChangeEvent } from 'react';
+
+/**
+ * Loads the ?session_id=SESSION_ID param from the <script> tag. Unfortunately
+ * import.meta.url doesn't have live URL params so we need to scrape the DOM.
+ */
+function loadSessionId(): string | undefined {
+  let sessionId;
+  Array.from(document.scripts).every((scriptTag) => {
+    if (scriptTag.src.includes('session_id')) {
+      sessionId = new URL(scriptTag.src).searchParams.get('session_id');
+      return false;
+    }
+    return true;
+  });
+  return sessionId;
```

Comment:
The use of `#every` here doesn't seem appropriate, since we're not doing anything with the result. `#forEach` could work, but I think we could also adjust this to use `#find` instead?

```suggestion
  const scriptTag = Array.from(document.scripts).find((scriptTag) => {
    return scriptTag.src.includes('session_id');
  });
  
  if (scriptTag) {
    return new URL(scriptTag.src).searchParams.get('session_id');
  }
```

Reply:
> I picked `#every` because I wanted a `#forEach` that I could early return from. `#find` seems like the right call, and I'm going to see if [currentScript](https://github.com/18F/identity-idp/pull/7018#discussion_r978930438) works so we don't have to search the DOM

Ah, that's a good point on the early return, but yeah, `#find` would work that way as well.

Reply:
`currentScript` worked, no need to iterate. 17e59691cc9b0a52598dcd253902ed137c793b17